addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -797,7 +797,20 @@ options.registry.OptionsTab = options.Class.extend({
     //--------------------------------------------------------------------------
     // Options
     //--------------------------------------------------------------------------
-
+    /**
+     * @see this.selectClass for parameters
+     */
+    async customizeWebsiteGrayParam(previewMode, widgetValue, params) {
+        if (!previewMode) {
+            await this._customizeWebsite(previewMode, widgetValue, params, 'variable');
+            return;
+        }
+        this.$el.find("span#grays").each((_, e) => {
```

Comment:
@qsm-odoo
This part of the code is to update the preview of the gray palette.
I have a doubt about this any suggestions!

Reply:
The UI update is to be done... in the `updateUI` method :wink: The option methods are to customize the snippet or the option's state: in this case, updating an internal variable for hue / saturation (`this.grayHue = ...`) (and maybe already compute the new grays). Then in updateUI, which is called automatically, you can check those internal variable and adapt the UI accordingly.

Reply:
@oussamamessaoudi You can reply to comments on PR directly instead of linking me the PR in discord and asking me there. It makes it hard to follow Github discussions afterwards otherwise :wink:

Mmh indeed my bad, for preview mode you can indeed update the UI directly in that method then :+1: